### ISSUE TITLE
IPv6 Compliance RFC4861: Prefix Information Option Processing, On-link Flag (Hosts Only) [v6LC.2.1.3]

### ISSUE THREAD

**Issue by LiveFreeAndRoam** [systemd/systemd#28435]
### systemd version the issue has been seen with

250

### Used distribution

Embedded Linux - Debian-based

### Linux kernel version used

5.15.71-rt51+gc36e774d0d9a

### CPU architectures issue was seen on

aarch64

### Component

systemd-networkd

### Expected behaviour you didn't see

Networkd MUST NOT interpret the Prefix Information `OnLink (L)` flag when it is clear.

RFC4861 - Extract from Page 54-55:
```
   Prefix Information options that have the "on-link" (L) flag set
   indicate a prefix identifying a range of addresses that should be
   considered on-link.  Note, however, that a Prefix Information option
   with the on-link flag set to zero conveys no information concerning
   on-link determination and MUST NOT be interpreted to mean that
   addresses covered by the prefix are off-link.  The only way to cancel
   a previous on-link indication is to advertise that prefix with the
   L-bit set and the Lifetime set to zero.
```

### Unexpected behaviour you saw

Networkd interpreted the Prefix Information with a cleared `OnLink (L)` flag, when it MUST ignore i t.

### Steps to reproduce the problem

The attached file contains a `replay` folder.  You can replay those protocol exchanges using:

```
$ sudo /usr/local/bin/tcpreplay -o -i ens224 v6LC_2_1_03.pcapng 
```

After each protocol exchange, you can examine the routing table.

#### Packet 1: Router LL: RA Prefix: 2001:2:1000 (On-Link=1)
```
$ ip -6 r get 2001:2:0:1000::
2001:2:0:1000:: dev eno0 proto ra src 2001:2:0:1000:9640:c9ff:fed6:dcd7 metric 10 pref medium
```
#### Packet 2: Router LL: RA Prefix: 2001:2:1000 (On-Link=0)
```
$ ip -6 r get 2001:2:0:1000::
2001:2:0:1000:: via fe80::200:10ff:fe10:1060 dev eno0 proto ra src 2001:2:0:1000:9640:c9ff:fed6:dcd7 metric 10 pref medium
```

According to RFC 4861, section 6.3.4, Packet 2 ought _not_ have affected the routing table, but clearly it has.

[2_1_3 - Prefix Information Option Processing, On-link Flag.zip](https://github.com/systemd/systemd/files/12074222/2_1_3.-.Prefix.Information.Option.Processing.On-link.Flag.zip)

**Comment by yuwata** [comment]
Fix is waiting in #31177.

**Comment by LiveFreeAndRoam** [comment]
I'll be able to validate this on Tuesday, next week.

**Comment by LiveFreeAndRoam** [comment]
I retested this on 253.5 and also from ToT.  I see it passing with both builds.  So, this one was fixed sometime between 250 and 253.5.  This one is verified.

**Comment by yuwata** [comment]
It seems I did something wrong in #31177...
